crates/subspace-farmer/src/cluster/controller/farms/stream_map.rs

```diff
@@ -45,9 +45,9 @@ where
         }
     }
 
-    /// Drops the task if there is already a future for the given `index` in `in_progress`.
+    /// Skip the task if there is already a future for the given `index` in `in_progress`.
     /// Returns `true` if the task is added to `in_progress`, `false` otherwise.
-    pub(super) fn drop_if_in_progress(&mut self, index: Index, fut: TaskFuture<'a, R>) -> bool {
+    pub(super) fn skip_if_in_progress(&mut self, index: Index, fut: TaskFuture<'a, R>) -> bool {
```

Comment:
The name here could be much better. The fact that the primary goal of the method appears to be to skip is misleading.

What about `add_if_not_in_progress`? It makes much more sense to me that way.